src/test/frontend/deploy/deployfromsettings_controller_test.js

```diff
@@ -336,4 +336,39 @@ describe('DeployFromSettings controller', () => {
       expect(ctrl.imagePullSecret).toEqual('');
     });
   });
+
+  /**
+   * Test for the name validation pattern (namePattern)
+   *
+   * The pattern matches strings that start and end with a lowercase letter
+   * and contains only lowercase letters and hyphens.
+   * Full pattern used for validation:
+   *
+   * ^[a-z]+(-[a-z]+)*$
+   */
```

Comment:
@digitalfishpond 
1st: I just would like to see a test-case that proves your claim. :) Make it visible for others.
2nd: I would like to see a test-case that visualizes the restrictions. Just for everyone to understand what kind of limitations the design causes.